src/coreclr/jit/gentree.h

```diff
@@ -98,9 +98,25 @@ enum genTreeKinds
     GTK_COMMUTE = 0x08,   // commutative  operator
     GTK_EXOP    = 0x10,   // Indicates that an oper for a node type that extends GenTreeOp (or GenTreeUnOp)
                           // by adding non-node fields to unary or binary operator.
-    GTK_NOVALUE   = 0x20, // node does not produce a value
-    GTK_NOTLIR    = 0x40, // node is not allowed in LIR
-    GTK_NOCONTAIN = 0x80, // this node is a value, but may not be contained
+    GTK_NOVALUE = 0x20,   // node does not produce a value
+
+    GTK_MASK = 0xFF
+};
+
+// The following enum defines a set of bit flags that describe opers for the purposes
+// of DEBUG-only checks. This is separate from the above "GenTreeOperKind"s to avoid
+// making the table for those larger in Release builds. However, it resides in the same
+// "namespace" and so all values here must be distinct from those in "GenTreeOperKind".
+//
+enum GenTreeDebugOperKind
+{
+    DBK_FIRST_FLAG = GTK_MASK + 1,
+
+    DBK_NOTHIR    = DBK_FIRST_FLAG,      // This oper is not supported in HIR (before rationalization).
```

Comment:
We also have "early HIR" in the compiler, i. e. HIR before morph: `GT_FIELD`, `GT_PUTARG_TYPE`, `GT_RET_EXPR`, `GT_RUNTIMELOOKUP`, `GT_CNS_STR`, `GT_INDEX`, so something like `DBK_EHIR` can now be easily added, if people feel like it is valuable (I personally do not see a lot of value).